grpc.go

```diff
@@ -768,17 +768,19 @@ func (a *agentGRPC) WaitProcess(ctx context.Context, req *pb.WaitProcessRequest)
 		return &pb.WaitProcessResponse{}, err
 	}
 
-	defer func() {
+	defer proc.Do(func() {
 		proc.closePostExitFDs()
 		ctr.deleteProcess(proc.id)
-	}()
+	})
 
 	// Using helper function wait() to deal with the subreaper.
 	libContProcess := (*reaperLibcontainerProcess)(&(proc.process))
-	exitCode, err := a.sandbox.subreaper.wait(proc.exitCodeCh, libContProcess)
-	if err != nil {
-		return &pb.WaitProcessResponse{}, err
-	}
+	exitCode, _ := a.sandbox.subreaper.wait(proc.exitCodeCh, libContProcess)
```

Comment:
Why is the error return ignored?

Reply:
Even though that might be technically correct *now*, I'd still check the result since if the wait function changes in the future, we might forget to update this part of the code.

Reply:
Yeah, I can return it back. : )